header.go

```diff
@@ -1620,6 +1620,7 @@ func (h *RequestHeader) parse(buf []byte) (int, error) {
 		if err != nil {
 			return 0, err
 		}
+		h.rawHeaders = buf[m : m+n]
```

Comment:
I don't think this is going to work. `parse` is called from [tryRead](https://github.com/erikdubbelboer/fasthttp/blob/516460b0797030211cb189fd97da6a5735818225/header.go#L1289-L1290) where it's buffer argument comes from [mustPeekBuffered](https://github.com/erikdubbelboer/fasthttp/blob/516460b0797030211cb189fd97da6a5735818225/header.go#L1360-L1361) which as you can see gets its return value from [bufio.Reader.Peek](https://golang.org/pkg/bufio/#Reader.Peek) which will change its return value after a next read (for example for the body) making `rawHeaders` contain invalid data.

I haven't tested this myself but I see your tests don't cover this case. Can you add a test here a whole POST request including a body of at least 4kb is parsed instead of just the RequestHeader.

Reply:
Thanks! I have just followed the same trail literally minutes ago, will add a test shortly.